README.md
TODO: This package is to show widgets in toast form
## Features

TODO: It has features to show any widget in toast form

## Getting started
<h1>Widget Toast</h1>

TODO: Just import package, call init method in initState and call static method to show toast

## Usage

TODO: it contains sample example whose image is given below.

```dart



import 'package:flutter/material.dart';
import 'package:widget_toast/widget_toast.dart';



class ToastOverlay extends StatefulWidget {
  ToastOverlay({Key? key}) : super(key: key);

  @override
  State<ToastOverlay> createState() => _ToastOverlayState();
}

class _ToastOverlayState extends State<ToastOverlay> {
  @override
  void initState() {
    // TODO: implement initState
    ToastWidget().init(context);
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Container(
          margin: const EdgeInsets.only(top: 300),
          height: 70,
          child: ListView(
            reverse: true,
            shrinkWrap: true,
            scrollDirection: Axis.horizontal,
            
            children: [
              const SizedBox(
                width: 50,
              ),
              ElevatedButton(
                  style: _buttonStyle(Colors.pink.shade300),
                  onPressed: () {
                    ToastUtil.showWidgetAsToast(
                        Image.asset(
                          paths[1],
                          width: 100,
                        ),
                        durationInMilliseconds: 1500,
                        backgroundColor: Colors.red,
                        toastPosition: ToastPosition.bottom);
                  },
                  child: const Text('Show Widget as Toast')),
              const SizedBox(
                width: 50,
              ),
              ElevatedButton(
                  style: _buttonStyle(Colors.red.shade300),
                  onPressed: () {
                    ToastUtil.showMessageToastWithIcon("Toast & icon on Top",
                        durationInMilliseconds: 1500,
                        backgroundColor: Colors.red,
                        toastPosition: ToastPosition.top);
                  },
                  child: const Text('Show Overlay Toast & icon on Top')),
              const SizedBox(
                width: 50,
              ),
              ElevatedButton(
                  style: _buttonStyle(Colors.green.shade300),
                  onPressed: () {
                    ToastUtil.showMessageToastWithIcon(
                        "Toast & icon at Center ",
                        durationInMilliseconds: 1500,
                        backgroundColor: Colors.green,
                        toastPosition: ToastPosition.middle);
                  },
                  child: const Text('Show Overlay Toast and icon at Center')),
              const SizedBox(
                width: 50,
              ),
              ElevatedButton(
                  style: _buttonStyle(Colors.purple.shade300),
                  onPressed: () {
                    ToastUtil.showMessageToastWithIcon("Toast & icon at bottom",
                        durationInMilliseconds: 1500,
                        backgroundColor: Colors.purple,
                        toastPosition: ToastPosition.bottom);
                  },
                  child: const Text('Show Overlay Toast and icon at Bottom')),
              const SizedBox(
                width: 50,
              ),
              ElevatedButton(
                  style: _buttonStyle(Colors.black),
                  onPressed: () {
                    ToastUtil.showMessageToast("This is toast",
                        durationInMilliseconds: 1500);
                  },
                  child: const Text('Show Overlay Toast')),
              const SizedBox(
                width: 50,
              ),
            ],
          ),
        ),
      ),
    );
  }

  ButtonStyle _buttonStyle(Color? color) {
    return ButtonStyle(
        backgroundColor:
            MaterialStateProperty.all<Color>(color ?? Colors.purple.shade500));
  }
}


```

<table>
  <tr>
    <td><img src="https://github.com/sdycode/widget_toast/blob/main/assets/ezgif.com-gif-maker.gif?raw=true" alt = "" width="200"></td>
     <td><img src="https://github.com/sdycode/widget_toast/blob/main/assets/ezgif.com-gif-maker%20(1).gif?raw=true" alt = "" width="200"></td>
      <td><img src="https://github.com/sdycode/widget_toast/blob/main/assets/ezgif.com-gif-maker%20(2).gif?raw=true" alt = "" width="200"></td>
  </tr>
  
 
  <tr>
   <td><img src="https://github.com/sdycode/widget_toast/blob/main/assets/ezgif.com-gif-maker%20(3).gif?raw=true" alt = "" width="200"></td>
    <td><img src="https://github.com/sdycode/widget_toast/blob/main/assets/ezgif.com-gif-maker%20(5).gif?raw=true" alt = "" width="200"></td>
     <td><img src="https://github.com/sdycode/widget_toast/blob/main/assets/Screenrecorder-2022-08-14-22-21-17-629(0).gif?raw=true" alt = "" width="200"></td>
  </tr>
</table>


## Additional information

TODO: In future more animations will be added to show customised toast


CHANGELOG.md
## 0.0.1

* TODO: Describe initial release.

## 0.0.2

* Gifs positioned changed

## 0.0.3

* Good quality gifs uploaded
  


test/widget_toast_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:widget_toast/widget_toast.dart';

void main() {
  test('adds one to input values', () {
   
  });
}


lib/src/toast_widgets/icontoast.dart
import 'package:flutter/material.dart';

class ToastIconWidget extends StatelessWidget {
  final value;
  final msg;
  final durationInMilliseconds;
  final icon;
  final bgColor;
  final toastPosition;
  ToastIconWidget(this.value, this.msg, this.durationInMilliseconds, this.icon,
      this.bgColor, this.toastPosition,
      {Key? key})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Material(
      type: MaterialType.transparency,
      child: Opacity(
        opacity: value,
        child: Container(
            padding: EdgeInsets.all(8),
            decoration: BoxDecoration(
                color: bgColor, borderRadius: BorderRadius.circular(6)),
            child: Row(
              children: [
                icon,
                Expanded(
                  child: Text(
                    msg,
                    textAlign: TextAlign.center,
                    style: TextStyle(color: Colors.white, fontSize: 24),
                  ),
                ),
              ],
            )),
      ),
    );
  }
}


lib/src/toast_widgets/textToast.dart
import 'package:flutter/material.dart';

class TextToast extends StatelessWidget {
  final value;
  final msg;
  final durationInMilliseconds;
  final icon;
  final bgColor;
  final toastPosition;
  TextToast(this.value, this.msg, this.durationInMilliseconds, this.icon,
      this.bgColor, this.toastPosition,
      {Key? key})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Material(
      type: MaterialType.transparency,
      child: Opacity(
        opacity: value,
        child: Container(
          padding: EdgeInsets.all(8),
          decoration: BoxDecoration(
              color: bgColor, borderRadius: BorderRadius.circular(6)),
          child: const Center(
              child: Text(
            "Toast showed using Overlay",
            textAlign: TextAlign.center,
            style: TextStyle(color: Colors.white, fontSize: 24),
          )),
        ),
      ),
    );
  }
}


lib/src/example/example.dart


import 'package:flutter/material.dart';
import 'package:widget_toast/widget_toast.dart';



class ToastOverlay extends StatefulWidget {
  ToastOverlay({Key? key}) : super(key: key);

  @override
  State<ToastOverlay> createState() => _ToastOverlayState();
}

class _ToastOverlayState extends State<ToastOverlay> {
  @override
  void initState() {
    // TODO: implement initState
    ToastWidget().init(context);
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Container(
          margin: const EdgeInsets.only(top: 300),
          height: 70,
          child: ListView(
            reverse: true,
            shrinkWrap: true,
            scrollDirection: Axis.horizontal,
            
            children: [
              const SizedBox(
                width: 50,
              ),
              ElevatedButton(
                  style: _buttonStyle(Colors.pink.shade300),
                  onPressed: () {
                    ToastUtil.showWidgetAsToast(
                        Image.asset(
                          paths[1],
                          width: 100,
                        ),
                        durationInMilliseconds: 1500,
                        backgroundColor: Colors.red,
                        toastPosition: ToastPosition.bottom);
                  },
                  child: const Text('Show Widget as Toast')),
              const SizedBox(
                width: 50,
              ),
              ElevatedButton(
                  style: _buttonStyle(Colors.red.shade300),
                  onPressed: () {
                    ToastUtil.showMessageToastWithIcon("Toast & icon on Top",
                        durationInMilliseconds: 1500,
                        backgroundColor: Colors.red,
                        toastPosition: ToastPosition.top);
                  },
                  child: const Text('Show Overlay Toast & icon on Top')),
              const SizedBox(
                width: 50,
              ),
              ElevatedButton(
                  style: _buttonStyle(Colors.green.shade300),
                  onPressed: () {
                    ToastUtil.showMessageToastWithIcon(
                        "Toast & icon at Center ",
                        durationInMilliseconds: 1500,
                        backgroundColor: Colors.green,
                        toastPosition: ToastPosition.middle);
                  },
                  child: const Text('Show Overlay Toast and icon at Center')),
              const SizedBox(
                width: 50,
              ),
              ElevatedButton(
                  style: _buttonStyle(Colors.purple.shade300),
                  onPressed: () {
                    ToastUtil.showMessageToastWithIcon("Toast & icon at bottom",
                        durationInMilliseconds: 1500,
                        backgroundColor: Colors.purple,
                        toastPosition: ToastPosition.bottom);
                  },
                  child: const Text('Show Overlay Toast and icon at Bottom')),
              const SizedBox(
                width: 50,
              ),
              ElevatedButton(
                  style: _buttonStyle(Colors.black),
                  onPressed: () {
                    ToastUtil.showMessageToast("This is toast",
                        durationInMilliseconds: 1500);
                  },
                  child: const Text('Show Overlay Toast')),
              const SizedBox(
                width: 50,
              ),
            ],
          ),
        ),
      ),
    );
  }

  ButtonStyle _buttonStyle(Color? color) {
    return ButtonStyle(
        backgroundColor:
            MaterialStateProperty.all<Color>(color ?? Colors.purple.shade500));
  }
}

/*
Show toast in flutter without using Flutter Plugin.
Create toast using overlay widget  with animation.
Change opacity and position to get effect of Toast

Source code : https://gist.github.com/sdycode/6f580f49afa779e01230dbe1640c9eb9

Music used in Video :
Music credit goes to Epidemic Sound  :
Turmoil : Rasmus Faber
https://www.epidemicsound.com/artists/rasmus-faber/
*/
